tests/providers/amazon/aws/hooks/test_eks.py

```diff
@@ -0,0 +1,872 @@
+            self.existing_cluster_name, self.nonexistent_cluster_name = random_names(
+                name_list=self.cluster_names
+            )
+
+            # Generate a list of the Cluster attributes to be tested when validating results.
+            self.attributes_to_test: List[Tuple] = attributes_to_test(
+                inputs=ClusterInputs, cluster_name=self.existing_cluster_name
+            )
+
+    def _execute(
+        count: Optional[int] = 1, minimal: Optional[bool] = True
+    ) -> Tuple[EKSHook, ClusterTestDataFactory]:
+        return eks_hook, ClusterTestDataFactory(count=count, minimal=minimal)
+
+    mock_eks().start()
+    eks_hook = EKSHook(
+        aws_conn_id=CONN_ID,
+        region_name=REGION,
+    )
+    yield _execute
+    mock_eks().stop()
+
+
+@pytest.fixture(scope="function")
+def nodegroup_builder(cluster_builder):
+    """A fixture to generate a batch of EKSManaged Nodegroups on the mocked backend for testing."""
+
+    class NodegroupTestDataFactory:
+        """A Factory class for building the Cluster objects."""
+
+        def __init__(self, count: int, minimal: bool) -> None:
+            self.cluster_name: str = cluster.existing_cluster_name
+
+            # Generate 'count' number of random Nodegroup objects.
+            self.nodegroup_names: List[str] = generate_nodegroups(
+                eks_hook=eks_hook, cluster_name=self.cluster_name, num_nodegroups=count, minimal=minimal
+            )
+
+            # Get the name of the first generated Nodegroup.
+            self.first_name: str = self.nodegroup_names[0]
+
+            # Collect the output of describe_nodegroup() for the first Nodegroup.
+            self.nodegroup_describe_output: Dict = json.loads(
+                eks_hook.describe_nodegroup(clusterName=self.cluster_name, nodegroupName=self.first_name)
+            )[ResponseAttribute.NODEGROUP]
+
+            # Pick a random Nodegroup name from the list and a name guaranteed not to be on the list.
+            self.existing_nodegroup_name, self.nonexistent_nodegroup_name = random_names(
+                name_list=self.nodegroup_names
+            )
+            _, self.nonexistent_cluster_name = random_names(name_list=[self.cluster_name])
+
+            # Generate a list of the Nodegroup attributes to be tested when validating results.
+            self.attributes_to_test: List[Tuple] = attributes_to_test(
+                inputs=NodegroupInputs,
+                cluster_name=self.cluster_name,
+                nodegroup_name=self.existing_nodegroup_name,
+            )
+
+    def _execute(
+        count: Optional[int] = 1, minimal: Optional[bool] = True
+    ) -> Tuple[EKSHook, NodegroupTestDataFactory]:
+        return eks_hook, NodegroupTestDataFactory(count=count, minimal=minimal)
+
+    eks_hook, cluster = cluster_builder()
+    return _execute
+
+
+@unittest.skipIf(mock_eks is None, reason=PACKAGE_NOT_PRESENT_MSG)
```

Comment:
Since this is not a unittest subclass
```suggestion
@pytest.mark.skipif(mock_eks is None, reason=PACKAGE_NOT_PRESENT_MSG)
```

Reply:
Thanks, I'll incorporate that in my coming revision.